src/lib/structures/Logger.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function */
+
+import EventEmitter from 'events';
+import JSLogger, { ILogger, ILoggerOpts, ILogHandler, ILogLevel } from 'js-logger';
+import { Util } from '../util/Util';
+import type { GlobalLogger } from 'js-logger';
+import type { AutocompleteContext } from './contexts/AutocompleteContext';
+import type { CommandContext } from './contexts/CommandContext';
+import type { ComponentContext } from './contexts/ComponentContext';
+import type { Command } from './Command';
+import type { Component } from './Component';
+import type { Listener } from './Listener';
+import type { Plugin } from './Plugin';
+
+export enum LoggerEvents {
+	'HANDLER_RUN' = 'handlerRun',
+	'HANDLER_ERROR' = 'handlerError',
+	'COMMAND_REGISTERED' = 'commandRegistered',
+	'COMMAND_UNREGISTERED' = 'commandUnregistered',
+	'COMPONENT_REGISTERED' = 'componentRegistered',
+	'COMPONENT_UNREGISTERED' = 'componentUnregistered',
+	'LISTENER_REGISTERED' = 'listenerRegistered',
+	'LISTENER_UNREGISTERED' = 'listenerUnregistered',
+	'PLUGIN_REGISTERED' = 'pluginRegistered',
+}
+
+export interface LoggerEventsInterface {
+	'handlerRun': (ctx: AutocompleteContext | CommandContext | ComponentContext) => void;
+	'handlerError': (ctx: AutocompleteContext | CommandContext | ComponentContext, error: any) => void;
+	'commandRegistered': (command: Command) => void;
+	'commandUnregistered': (command: Command) => void;
+	'componentRegistered': (component: Component) => void;
+	'componentUnregistered': (component: Component) => void;
+	'listenerRegistered': (listener: Listener) => void;
+	'listenerUnregistered': (listener: Listener) => void;
+	'pluginRegistered': (plugin: Plugin) => void;
+}
+
+export declare interface LoggerClass {
+	on<U extends keyof LoggerEventsInterface>(
+	  event: U, listener: LoggerEventsInterface[U]
+	): this;
+  
+	emit<U extends keyof LoggerEventsInterface>(
+	  event: U, ...args: Parameters<LoggerEventsInterface[U]>
+	): boolean;
+}
+
+export class LoggerClass extends EventEmitter implements GlobalLogger {
```

Comment:
I think the class should be called `Logger` and the instance `logger`. LoggerClass is a bit weird.

Reply:
Instance can certainly not be called `logger`, due to a minor change.

Reply:
Well i can rename `LoggerClass` to `ILogger` and instance just be `Logger`